gazebo/sensors/SensorManager.cc

```diff
@@ -132,6 +136,116 @@ void SensorManager::WaitForSensors(double _clk, double _dt)
   }
 }
 
+std::map<std::string, gazebo::common::Time> sensorsLastMeasurementTime;
+std::map<std::string, gazebo::common::Time> worldLastMeasurementTime;
+struct sensorPerformanceMetricsType
+{
+  double sensorRealUpdateRate;
+  double sensorSimUpdateRate;
+  double sensorFPS;
+};
+std::map<std::string, struct sensorPerformanceMetricsType> sensorPerformanceMetrics;
+/// \brief Publisher for run-time simulation performance metrics.
+transport::PublisherPtr performanceMetricsPub;
+transport::NodePtr node;
+
+void PublishPerformanceMetrics()
+{
+  if (!performanceMetricsPub || !performanceMetricsPub->HasConnections())
+  {
+    return;
+  }
+
+  physics::WorldPtr world = physics::get_world("default");
+
+  /// Outgoing run-time simulation performance metrics.
+  msgs::PerformanceMetrics performanceMetricsMsg;
+
+  // Real time factor
+  common::Time realTime = world->RealTime();
+  common::Time simTime = world->SimTime();
+  if (realTime == 0)
+    simTime = 0;
+  else
+    simTime = simTime / realTime;
+
+  if (simTime > 0)
+    performanceMetricsMsg.set_real_time_factor(simTime.Double());
```

Comment:
reviving the comment about this being the overall RTF for the entire simulation. It maybe more useful to display the current RTF, i.e. instantaneous or avg value over a window size. 

e.g. if I change run the simulation for 5 min and change the `real_time_update_rate` to 100 from 1000, the value published here takes a while to converge down to an RTF of ~0.1

Reply:
instantaneous RTF: https://github.com/osrf/gazebo/pull/2819/commits/66426aa94c607e97f8f4567143b709119e57fb84